### ISSUE TITLE
[Operator] Delete collector image version from the values.yaml

### ISSUE THREAD

**Issue by jdiegosierra** [open-telemetry/opentelemetry-helm-charts#1189]
Hello,

I noticed that currently _.Chart.AppVersion_ is the default value for the opentelemetry-operator.image
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/2f4e7a8f960366b961d621c9bc7c6ae389129af3/charts/opentelemetry-operator/templates/_helpers.tpl#L150

Keeping in mind that the operator and the collector should have the same version https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#opentelemetry-operator-vs-opentelemetry-collector. 

Why not using appVersion as default for collectorImage as well?

Or simply why the values.yaml has the collector image set? 
https://github.com/open-telemetry/opentelemetry-helm-charts/blob/2f4e7a8f960366b961d621c9bc7c6ae389129af3/charts/opentelemetry-operator/values.yaml#L45

If I'm not wrong, if we don't set the  _--collector-image flag_ https://github.com/open-telemetry/opentelemetry-helm-charts/blob/2f4e7a8f960366b961d621c9bc7c6ae389129af3/charts/opentelemetry-operator/templates/deployment.yaml#L51
The operator will take care of deploying a collector with the same version as itself, isn't it? So why is needed to set a value for the collector by default?

**Comment by TylerHelmuth** [comment]
> The operator will take care of deploying a collector with the same version as itself, isn't it? So why is needed to set a value for the collector by default?

I'd rather do this instead of reuse `appVersion`.

**Comment by jdiegosierra** [comment]
> > The operator will take care of deploying a collector with the same version as itself, isn't it? So why is needed to set a value for the collector by default?
> 
> I'd rather do this instead of reuse `appVersion`.

Yeah, probably it makes more sense. So there is no reason to have a default value in the values.yaml, right? Do you agree to delete the default value?

**Comment by TylerHelmuth** [comment]
Yes I think that is a safe, un-breaking change that will make operator upgrades easier

**Comment by jdiegosierra** [comment]
I'm going to check if the operator works as expected in case of not using the flag `--collector-image`

**Comment by jdiegosierra** [comment]
> I'm going to check if the operator works as expected in case of not using the flag `--collector-image`

Seems that it works as expected, I deployed the operator setting an empty tag:
```yaml
collectorImage:
    repository: ghcr.io/otel/opentelemetry-collector-contrib
    tag: ""
```
```bash
(k8s|kind-kind:otel-test) ~/Projects/Personal/opentelemetry-helm-charts/charts/opentelemetry-operator: k describe pod simplest-collector-54bb659675-m7jfg | grep Image
    Image:         ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector:0.99.0
    Image ID:      ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector@sha256:bba8474ea69d11f0450e38c50040efea47f4190723a9fb64cc9d44939b4ade4c
```

**Comment by jdiegosierra** [comment]
Oh, I was not aware of this https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1153.

**Comment by jdiegosierra** [comment]
Taking into account the https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1153 I tried with  `--collector-image=ghcr.io/otel/opentelemetry-collector-contrib` rather than  `--collector-image=ghcr.io/otel/opentelemetry-collector-contrib:0.99.0` and it doesn't work, so I proceed with closing the issue.

**Comment by jdiegosierra** [comment]
Taking into account the https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1153 I tried with --collector-image=ghcr.io/otel/opentelemetry-collector-contrib rather than --collector-image=ghcr.io/otel/opentelemetry-collector-contrib:0.99.0 and it doesn't work, so I proceed with closing the issue.

**Comment by jdiegosierra** [comment]
Taking into account the https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1153 I tried with --collector-image=ghcr.io/otel/opentelemetry-collector-contrib rather than --collector-image=ghcr.io/otel/opentelemetry-collector-contrib:0.99.0 and it doesn't work, so I proceed with closing the issue.

**Comment by jdiegosierra** [comment]
I noticed that I was using a wrong collector URL to test the flag. I used `ghcr.io/otel/opentelemetry-collector-contrib` rather than `otel/opentelemetry-collector` or `otel/opentelemetry-collector-contrib`. Now it works properly. So I think it is safe to delete the tag version.
